### PR TITLE
Fix parsing issue in transaction search

### PR DESCRIPTION
**Pull request by melindaloubser1**
- fix from and to date parsing
- localize dates
- store dates in isoformat, and store formatted dates seperately

### CODE REVIEW

File: actions/parsing.py [hunk 1]
```diff
@@ -55,7 +58,8 @@ def parse_duckling_time_as_interval(
         return make_interval_from_value_duckling_time(timeinfo)
 
 
-def format_time_by_grain(time, grain=None):
+def format_isotime_by_grain(isotime, grain=None):
+    time = parser.isoparse(isotime)
```
Comment (b-quachtran) [inline actions/parsing.py:62]: `time` is a standard python library, so suggesting a rename here

```suggestion
    value = parser.isoparse(isotime)
```

File: actions/parsing.py [hunk 2]
```diff
@@ -64,18 +68,20 @@ def format_time_by_grain(time, grain=None):
         "year": "%Y",
     }
     timeformat = grain_format.get(grain, "%I:%M%p, %A %b %d, %Y")
-    return time.strftime(timeformat)
+    time_formatted = time.strftime(timeformat)
```
Comment (b-quachtran) [inline actions/parsing.py:71]:
```suggestion
    time_formatted = value.strftime(timeformat)
```

File: actions/parsing.py [hunk 3]
```diff
@@ -64,18 +68,20 @@ def format_time_by_grain(time, grain=None):
...
+    return time_formatted
 
 
 def parse_duckling_time(
     timeentity: Dict[Text, Any]
 ) -> Optional[Dict[Text, Any]]:
     timeinfo = timeentity.get("additional_info", {})
     if timeinfo.get("type") == "value":
-        value = timeinfo.get("value")
+        time = timeinfo.get("value")
```
Comment (b-quachtran) [inline actions/parsing.py:80]:
```suggestion
        value = timeinfo.get("value")
```

File: actions/parsing.py [hunk 4]
```diff
@@ -64,18 +68,20 @@ def format_time_by_grain(time, grain=None):
...
         grain = timeinfo.get("grain")
         parsedtime = {
-            "time": format_time_by_grain(parser.isoparse(value), grain),
+            "time": time,
```
Comment (b-quachtran) [inline actions/parsing.py:83]:
```suggestion
            "time": value,
```

File: actions/parsing.py [hunk 5]
```diff
@@ -64,18 +68,20 @@ def format_time_by_grain(time, grain=None):
...
+            "time_formatted": format_isotime_by_grain(time, grain),
```
Comment (b-quachtran) [inline actions/parsing.py:84]:
```suggestion
            "time_formatted": format_isotime_by_grain(value, grain),
```

File: actions/parsing.py (hunk 1 above)
Comment (melindaloubser1) [reply actions/parsing.py:62]: ooh good point

**Pull request merged**